src/clj/puppetlabs/services/master/master_core.clj

```diff
@@ -14,13 +18,62 @@
   "v3")
 
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;; Private
+
+(defn class-info-from-jruby->class-info-for-json
+  "Convert a class info map received from the jruby service into an
+  appropriate map for use in serializing an environment_classes response to
+  JSON"
+  [info-from-jruby environment]
+  (->> info-from-jruby
+       (map #(hash-map :path (key %) :classes (val %)))
+       (sort-by :path)
+       (into [])
+       (hash-map :name environment :files)))
+
+(defn get-environment-class-info
+  "Middleware function for constructing a Ring response from an incoming
+  request for environment_classes information"
+  [jruby-service]
+  (fn [request]
+    (if-let [environment (jruby-request/get-environment-from-request request)]
+      (if (re-matches #"^\w+$" environment)
+        (if-let [class-info
+                 (jruby-protocol/get-environment-class-info jruby-service
+                                                            (:jruby-instance
+                                                             request)
+                                                            environment)]
+          (-> class-info
+              (class-info-from-jruby->class-info-for-json environment)
+              (cheshire/generate-string)
+              (rr/response)
+              (rr/content-type "application/json"))
+          (rr/not-found (str "Could not find environment '" environment "'")))
+        (jruby-request/throw-bad-request!
+         (str
+          "The environment must be purely alphanumeric, not '"
+          environment
+          "'")))
```

Comment:
If I did this with a Clojure `cond` instead of the nested if statements would that seem a little more readable - or, even then, would you suggest still moving some of this into another helper function?